lib/semantics/check-do-concurrent.cc

```diff
@@ -450,8 +450,35 @@ class DoConcurrentContext {
 private:
   bool ExpressionHasTypeCategory(const evaluate::GenericExprWrapper &expr,
       const common::TypeCategory &type) {
-    // TODO - implement
-    return false;
+    return std::visit(
```

Comment:
```
auto dynamicType{expr.v.GetType()};
return dynamicType.has_value() && dynamicType->category == type;
```

Reply:
Thanks. I figured there must be a simpler way, but ...